src/jennifer/adapter/mysql/command_interface.cr

```diff
@@ -17,7 +17,8 @@ module Jennifer
 
       def generate_schema
         options = ["-u", config.user, "--no-data", "-h", config.host, "--skip-lock-tables", config.db]
-        options += ["--password=#{config.password}"] if !config.password.empty?
+        options += ["--password='#{config.password}'"] unless config.password.empty?
+        options += ["--port=#{config.port}"] unless config.port.zero?
```

Comment:
sorry, it isn't so obvious with the "blank" port value - in the initializer it is set to `-1` by default so this should be `if config.port > -1` 😅

Reply:
Update.
Out of curiosity, why did you set the port value -1 in config.cr, not keep it to 0 as default?

Reply:
That was a long time ago. I believe I decided that `-1` value would explicitly show to developer that something is going wrong when it accidentally appear in the connection string